### DISCUSSION TITLE
Pure USB-Midi implementation

### DISCUSSION THREAD

**Discussion by TuriSc** [marcel-licence/samd21_mini_synth#4] [Show and tell]
Hi. I forked your code and tweaked it to make a fully portable device that can power a small USB Midi keyboard.
[github.com/TuriSc/portami](https://github.com/TuriSc/portami)
The little thing has an onboard speaker and an audio out jack.

Thank you for your amazing synth!

**Comment by marcel-licence** [comment]
Hi,
thank you for sharing. Did you also observed NoteOn/Off messages losses?
Did you change something to fix it?

Best regards,
Marcel

**Comment by TuriSc** [comment]
I did encounter one glitch: when the controller gets overloaded with Midi messages it sometimes does not register a NoteOff event, and the note gets stuck until its key is pressed and released again. I did not address the issue but at the next iteration I will surely have a look.

**Reply by marcel-licence** [reply]
Hm, I had discovered a problem on the hardware layer. I was reading zeros sometimes from the internal memory. Maybe something has been fixed in the meantime by the USB host library. I should check it out.
By the way; nice work! =)
The organ project should be also capable of running on the SAMD21. I didn't test it for a long time
